src/components/radio-button/radio-button.pw.tsx

```diff
@@ -0,0 +1,815 @@
+        );
+        const radiobuttonGroupLegendElement = await radiobuttonGroupLegend(
+          page
+        );
+        await expect(radiobuttonGroupLegendElement).toHaveCSS(
+          "padding-right",
+          padding
+        );
+      });
+    }
+  );
+
+  (["top", "bottom", "left", "right"] as const).forEach((position) => {
+    test(`should render RadioButtonGroup component with tooltip positioned to the ${position}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <RadioButtonGroupComponent
+          error="Something is wrong"
+          tooltipPosition={position}
+        />
+      );
+      const radiobuttonGroupIconElement = await radiobuttonGroupIcon(page);
+      await radiobuttonGroupIconElement.hover();
+      const tooltipPreviewElement = await tooltipPreview(page);
+      await expect(tooltipPreviewElement).toHaveText("Something is wrong");
+      await expect(tooltipPreviewElement).toHaveAttribute(
+        "data-placement",
+        position
+      );
+    });
+  });
+
+  test("should render RadioButtonGroup component as a required field", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<RadioButtonGroupComponent required />);
+    await verifyRequiredAsteriskForLegend(page);
+  });
+
+  test("should render RadioButtonGroup component with radiobuttons inline", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<RadioButtonGroupComponent inline />);
+    const firstRadiobuttonElement = await radiobutton(page, 0);
+    await expect(firstRadiobuttonElement).toHaveCSS("margin-bottom", "12px");
+    await expect(firstRadiobuttonElement).toHaveCSS("margin-left", "0px");
+
+    const secondRadiobuttonElement = await radiobutton(page, 1);
+    await expect(secondRadiobuttonElement).toHaveCSS("margin-bottom", "12px");
+    await expect(secondRadiobuttonElement).toHaveCSS("margin-left", "32px");
+
+    const thirdRadiobuttonElement = await radiobutton(page, 2);
+    await expect(thirdRadiobuttonElement).toHaveCSS("margin-bottom", "0px");
+    await expect(thirdRadiobuttonElement).toHaveCSS("margin-left", "32px");
+  });
+
+  ([
+    [399, "left"],
+    [400, "left"],
+    [401, "none"],
+  ] as const).forEach(([breakpoint, float]) => {
+    test(`should check RadioButtonGroup legend is ${breakpoint} with adaptiveLabelBreakpoint ${float} and viewport 400`, async ({
+      mount,
+      page,
+    }) => {
+      await page.setViewportSize({ width: 400, height: 300 });
+      await mount(
+        <RadioButtonGroupComponent
+          adaptiveLegendBreakpoint={breakpoint}
+          adaptiveSpacingBreakpoint={breakpoint}
+        />
+      );
+      const radiobuttonGroupLegendElement = await radiobuttonGroupLegend(page);
+      await expect(radiobuttonGroupLegendElement).toHaveCSS("float", float);
+    });
+  });
+});
+
+test.describe("Accessibility tests for RadioButton component", () => {
+  test("should pass accessibilty tests for RadioButton Default story", async ({
```

Comment:
**suggestion:** These test titles refer to `story` but we aren't using stories anymore for these tests. I've used `example` instead in some of my refactors but you're more than welcome to be creative and come up with a more appropriate title for these 😄